P0009/P0009.md

```diff
@@ -1944,7 +1946,7 @@ template<class OtherExtents>
 ```c++
 constexpr size_type required_span_size() const noexcept;
 ```
-* [7]{.pnum} *Returns:* The maximum of `extents().extent(r) * stride(r)` for all `r` in the range $[0,$ `Extents::rank()` $)$.
+* [7]{.pnum} *Returns:* If `(extents().extent(r) > 0) && (Extents::rank() > 0)` is true for all `r` in the range $[0,$ `Extents::rank()`, returns the maximum of `extents().extent(r) * stride(r)` for all `r` in the range $[0,$ `Extents::rank()` $)$. Otherwise, returns `Extents::rank() == 0 ? 1 : 0`.
```

Comment:
we say that additional returns in other places if preceded by a condition or Otherwise.

Reply:
actually we only say it one place that was what I looked at :-) I don't think we need a note though. for an arbitrary strided layout this is just how you do it since it doesn't come the product of extents, since this thing can be non-contiguous.

Reply:
Why don't we write that, though?
```
*[Note:* A rank-0 `mdspan` represents a single element.  Thus, `required_span_size()` is one instead of zero in that case. --*end note]*
```
There are also a couple issues with the suggested wording:

> If `(extents().extent(r) > 0) && (Extents::rank() > 0)` is true for all `r` in the range $[0,$ `Extents::rank()`, ....

I think there's a missing close parenthesis after `rank()`.

> If `(extents().extent(r) > 0) && (Extents::rank() > 0)` is true for all `r` in the range $[0,$ `Extents::rank()`, returns the maximum of `extents().extent(r) * stride(r)` for all `r` in the range $[0,$ `Extents::rank()` $)$. Otherwise, returns `Extents::rank() == 0 ? 1 : 0`.

I think the `If` is trivially true if `rank()` is zero, because the range is empty.  (The "If" condition doesn't get evaluated at all, because there are no `r` in the range.)